nodeadm/internal/api/status.go

```diff
@@ -64,3 +71,69 @@ func privateDNSNameAvailable(out *ec2.DescribeInstancesOutput) (bool, error) {
 	}
 	return aws.ToString(out.Reservations[0].Instances[0].PrivateDnsName) != "", nil
 }
+
+func getNetworkCardsDetails(ctx context.Context, imdsFunc func(ctx context.Context, prop imds.IMDSProperty) (string, error)) ([]NetworkCardDetails, error) {
+
+	allMacs, err := imdsFunc(ctx, "network/interfaces/macs/")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get network interfaces from imds: %w", err)
+	}
+
+	availableMacs := parseAvailableMacs(allMacs)
+	details := []NetworkCardDetails{}
+
+	for _, mac := range availableMacs {
+		cardDetails, err := getNetworkCardDetail(ctx, imdsFunc, mac)
+		if err != nil {
+			if isNotFoundError(err) {
```

Comment:
For the efa-only, can we also verify if local-ipv4s field is present but empty ? Suggestion we got from the efa-only team is to verify both local-ipv4s field is not present and if present it is empty to make sure it works if there are any changes in the imds fields